arcjet/index.ts

```diff
@@ -309,11 +312,51 @@ export type EmailOptions = {
   allowDomainLiteral?: boolean;
 };
 
+type DetectEntities<T> = (
+  tokens: string[],
+) => (ArcjetSensitiveInfoType | T | undefined)[];
+
+type SensitiveInfoOptionsAllow<
+  Detect extends DetectEntities<CustomEntities>,
+  CustomEntities extends string,
+> = {
+  allow: (
+    | ArcjetSensitiveInfoType
+    | Exclude<ReturnType<Detect>[number], undefined>
+  )[];
+  deny?: never;
+  contextWindowSize?: number;
+  mode?: ArcjetMode;
+  detect?: Detect;
+};
+
+type SensitiveInfoOptionsDeny<
+  Detect extends DetectEntities<CustomEntities>,
+  CustomEntities extends string,
+> = {
+  allow?: never;
+  deny: (
+    | ArcjetSensitiveInfoType
+    | Exclude<ReturnType<Detect>[number], undefined>
+  )[];
+  contextWindowSize?: number;
+  mode?: ArcjetMode;
+  detect?: Detect;
+};
+
+export type SensitiveInfoOptions<
+  Detect extends DetectEntities<CustomEntities>,
+  CustomEntities extends string,
+> =
+  | SensitiveInfoOptionsAllow<Detect, CustomEntities>
+  | SensitiveInfoOptionsDeny<Detect, CustomEntities>;
+
 const Priority = {
   Shield: 1,
   RateLimit: 2,
   BotDetection: 3,
   EmailValidation: 4,
+  SensitiveInfo: 5,
```

Comment:
If we have sensitive info then we don't want to do any further processing - the request is likely to be blocked, so we can save time especially when this can be done locally. So I'd move it to `SensitiveInfo: 1`